flang/lib/Lower/Runtime.cpp

```diff
@@ -219,3 +219,36 @@ void Fortran::lower::genDateAndTime(Fortran::lower::FirOpBuilder &builder,
         builder.convertWithSemantics(loc, std::get<1>(op), std::get<0>(op)));
   builder.create<fir::CallOp>(loc, callee, operands);
 }
+
+void Fortran::lower::genSystemClock(FirOpBuilder &builder, mlir::Location loc,
+                                    llvm::Optional<mlir::Value> count,
+                                    llvm::Optional<mlir::Value> countRate,
+                                    llvm::Optional<mlir::Value> countMax) {
+  auto callee = genRuntimeFunction<mkRTKey(SystemClock)>(loc, builder);
+  mlir::Type idxTy = builder.getIndexType();
+  mlir::Value zero;
+  auto handleOptionalArgs = [&](llvm::Optional<mlir::Value> arg,
```

Comment:
More work is needed here since `COUNT` and `COUNT_MAX` can be have any integer type and `COUNT_RATE` can have any integer or real type. The results from the runtime need to be converted (at the value level, not pointer level) to the.
Note that if it makes easier, I think you can split `SystemClock` runtime into `long long RTNAME(SystemClockCount)()`, `long long RTNAME(SystemClockRate)()` and `long long RTNAME(SystemClockMax)()` and only call the ones needed for the intrinsic call. This is not possible to split DATE_AND_TIME because all the results needs to come from the same time point, but here only COUNT is derived from a time clock read.